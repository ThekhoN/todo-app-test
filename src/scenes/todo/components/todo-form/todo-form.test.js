import React from "react";
import TodoForm from "./index";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

describe("TodoForm", () => {
  let wrapper, onSubmit;
  beforeEach(() => {
    onSubmit = jest.fn();
    wrapper = shallow(<TodoForm onSubmit={onSubmit} />);
  });

  it("renders a form with class todo-form", () => {
    expect(
      wrapper.contains(<form className="todo-form" onSubmit={onSubmit} />)
    );
  });

  it("onSubmit runs form submit", () => {
    wrapper.simulate("submit");
    expect(onSubmit).toBeCalled();
  });

  it("runs onSubmit on click submit button", () => {
    const submitButton = wrapper.find("button");
    submitButton.simulate("click");
    expect(onSubmit).toBeCalled();
  });
});
