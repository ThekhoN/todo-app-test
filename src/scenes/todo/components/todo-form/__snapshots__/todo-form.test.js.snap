// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoForm snapshots correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodoForm
    onSubmit={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <TodoInput />,
        <button
          onClick={[Function]}
          type="submit"
        >
          add todo
        </button>,
      ],
      "className": "todo-form",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "add todo",
          "onClick": [Function],
          "type": "submit",
        },
        "ref": null,
        "rendered": "add todo",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <TodoInput />,
          <button
            onClick={[Function]}
            type="submit"
          >
            add todo
          </button>,
        ],
        "className": "todo-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "add todo",
            "onClick": [Function],
            "type": "submit",
          },
          "ref": null,
          "rendered": "add todo",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
